### PR TITLE
[feature] HSTS headers in aws-single-page-static-site

### PR DESCRIPTION
**Pull request by palasha**
This module already enforces that the site is accessed with HTTPS via
the cloudfront `"redirect-to-https"` policy.

Since that's true, we can also set HSTS headers on all the responses
using a cloudfront lambda, that, I've conveniently created already in
https://github.com/chanzuckerberg/cztack/pull/243

This does that.

Test Plan:
push up PR
update connections-infra module ref to new commit
make apply
curl to see if it works

### CODE REVIEW

**Comment by palasha** [comment]
override function name to prevent collision with naked domain redirect in same component

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/30669aeb2b76cc5b5783f8d8a2c8a25739f2d561..ccb96a45b50691946da6a11d35e50977dfa8ee47).

File: aws-single-page-static-site/main.tf [hunk 1]
```diff
@@ -82,6 +82,17 @@ resource "aws_s3_bucket_public_access_block" "bucket" {
   restrict_public_buckets = true
 }
 
+module "security_headers_lambda" {
+  source = "git@github.com:chanzuckerberg/cztack//aws-lambda-edge-add-security-headers?ref=7ba56765f2401062fe26bc601920046aa7a739c7"
```
Comment (ryanking) [inline aws-single-page-static-site/main.tf:86]:
```suggestion
  source = "../aws-lambda-edge-add-security-headers"
```

**Review by ryanking — CHANGES_REQUESTED** [review]
1. we should reference the other cztack module via a relative path

2. do you think this is safe to include for all usage or this module? I think so but want to make sure that you have thought through it.

**Comment by palasha** [comment]
relative path to cztack module

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/cztack/compare/0482e73661830525f0add1e3a3285ef8e432a962..1ca8bc7624dbd2f7e73e2c70e58deb8550fa9750).

**Comment by palasha** [comment]
> do you think this is safe to include for all usage or this module?

My thought was that since this module already enforces https, that this should be safe / better for existing users of the module as well.

**Review by damien — APPROVED** [review]
Reviewed this and the associated HSTS lambda. Well done!

**Review by ryanking — APPROVED** [review]

**Pull request merged**